_user/rhealstone/irq-latency/main.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Rhealstone benchmark
+ *
+ * Message latency
+ *
+ * Copyright 2024 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdatomic.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/interrupt.h>
+#include <sys/mman.h>
+#include <sys/time.h>
+#include <sys/threads.h>
+
+#include <board_config.h>
+#include <rhcommon.h>
+
+
+#define BENCHMARKS 10000
+
+atomic_uint_least64_t benchEnd = 0;
```

Comment:
Atomic might give unwanted overhead, especially 64 bit one - we trigger the interrupt via SW, so should be fine to use plain `volatile`